### PR TITLE
Angular 7

### LINKED ISSUE #889: Update all dependencies to the lastest possible version
# User Story

### As a: 
site maintainer

### I want:
my application to have all the latest updates

### so that:
i have a secure and up to date application

## Acceptance criteria:
- [ ] Frontend uses Angular 7
- [ ] Other frontend dependencies (e.g. for testing) are up to date
- [ ] All API Dependencies are up to date

### PR DESCRIPTION
**Pull request by dboschm**
## Description:
Updated dependancies

Closes #889

## Improvements
- angular 7 and more new dependancies in the frontend
- updated to new dragula API

## Known Issues:
-

### CODE REVIEW

File: app/webFrontend/src/app/shared/services/data.service.ts [hunk 1]
```diff
@@ -509,7 +509,7 @@ export class ConfigService extends DataService {
 
   async getDownloadMaxFileSize () {
     const res = <IConfig><any> await this.readSingleItem('public/downloadMaxFileSize');
-    const _value =  Number.parseInt(res.value);
+    const _value = Number.parseInt(res.value, 10);
```
Comment (dboschm) [inline app/webFrontend/src/app/shared/services/data.service.ts]: after updating the linter there is a new rule "radix missing" I hope 10 is the correct radix here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -82,6 +82,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Secure `{put} /api/notificationSettings/` by using the `@CurrentUser` instead of allowing arbitrary id requests. [#1072](https://github.com/geli-lms/geli/issues/1072)
 - Minimize `NotificationSettingsController` attack surface by severely simplifying its routes. [#1072](https://github.com/geli-lms/geli/issues/1072)
 
+###Changed
```
Comment (kesselb) [inline CHANGELOG.md]: Please add this to Unreleased.

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Extend `ProgressController` `PUT` route to handle both creation and updates. [#1116](https://github.com/geli-lms/geli/issues/1116)
 - Refactor `ProgressController` unit tests in general. [#1116](https://github.com/geli-lms/geli/issues/1116)
 - Instead of a list of progress data, the `ProgressController` `GET` route now responds with a single progress object or an empty object if no data can be found. [#1116](https://github.com/geli-lms/geli/issues/1116)
+- Updated frontend to angular 7[#889](https://github.com/geli-lms/geli/issues/889)
```
Comment (kesselb) [inline CHANGELOG.md]:
```suggestion
- Update frontend to Angular 7[#889](https://github.com/geli-lms/geli/issues/889)
```

**Review by Gargamil — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -27,6 +27,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Refactor `MediaController` unit tests in general using the `TestHelper`. [#1196](https://github.com/geli-lms/geli/issues/1196)
 - Instead of a list of progress data, the `ProgressController` `GET` route now responds with a single progress object or an empty object if no data can be found. [#1116](https://github.com/geli-lms/geli/issues/1116)
 - `Directory` / `File` schemata and the corresponding interfaces now reference their `Course` as `_course` (analogous to the `Unit` schema). [#1196](https://github.com/geli-lms/geli/issues/1196)
+- Update frontend to Angular 7[#889](https://github.com/geli-lms/geli/issues/889)
```
Comment (kesselb) [inline CHANGELOG.md]:
```suggestion
- Update frontend to Angular 7. [#889](https://github.com/geli-lms/geli/issues/889)
```

**Comment by kesselb** [comment]
image
https://github.com/geli-lms/geli/pull/1143/commits/4a4c7bc99845fde805e37dfab0677f8928a26e3f

**Comment by kesselb** [comment]
image

Download dialog looks odd with angular 7

**Comment by kesselb** [comment]
It's not possible to download any course or assignment with this branch. I fixed the code warnings but ... @dboschm @PatrickSkowronek

**Comment by kesselb** [comment]
> Once you and all of your dependencies have updated to RxJS 6, remove rxjs-compat.

**Comment by kesselb** [comment]
> It's not possible to download any course or assignment with this branch. I fixed the code warnings but ... @dboschm @PatrickSkowronek

`<Blob><any>response.body` would work. Do you have a better suggestion? The response is already a blob but response.body has another return type. So you have to cast it to any and to blob than.

**Comment by kesselb** [comment]
> rxjs-5-to-6-migrate -p src/tsconfig.app.json

image

**Comment by torss** [comment]
> `<Blob><any>response.body` would work. Do you have a better suggestion? The response is already a blob but response.body has another return type. So you have to cast it to any and to blob than.

@kesselb I haven't looked into this too much, but it appears that https://angular.io/api/http/Response is deprecated anyway, and there now is https://angular.io/api/common/http/HttpResponse "including a typed response body", returned by https://angular.io/api/common/http/HttpClient `get` overload nr. 10 (` observe: 'response', responseType: 'blob'` as used in `BackendService.getDownload`, which in turn is used from `DownloadFileService.getFile`).

**Comment by kesselb** [comment]
> > `<Blob><any>response.body` would work. Do you have a better suggestion? The response is already a blob but response.body has another return type. So you have to cast it to any and to blob than.
> 
> @kesselb I haven't looked into this too much, but it appears that https://angular.io/api/http/Response is deprecated anyway, and there now is https://angular.io/api/common/http/HttpResponse "including a typed response body", returned by https://angular.io/api/common/http/HttpClient `get` overload nr. 10 (` observe: 'response', responseType: 'blob'` as used in `BackendService.getDownload`, which in turn is used from `DownloadFileService.getFile`).

way better ;)

**Comment by kesselb** [comment]
Breaking changes from 6.3 to 7.2

https://github.com/angular/material2/blob/master/CHANGELOG.md#breaking-changes
https://github.com/angular/material2/blob/master/CHANGELOG.md#breaking-changes-1
https://github.com/angular/material2/blob/master/CHANGELOG.md#breaking-changes-2
https://github.com/angular/material2/blob/master/CHANGELOG.md#breaking-changes-3
https://github.com/angular/material2/blob/master/CHANGELOG.md#breaking-changes-4

**Pull request closed without merging**